tests/rados/test_objectstore_block.py

```diff
@@ -0,0 +1,302 @@
+        )
+
+        obj_dump_single_, _ = acting_osd_node.exec_command(
+            sudo=True, cmd=cmd_fetch_obj_dump
+        )
+        obj_dump_single = json.loads(obj_dump_single_)
+        obj_dump_single_stats = obj_dump_single["stat"]
+        obj_dump_single_extents = obj_dump_single["onode"]["extents"]
+
+        log.debug(f"Object dump for {object_name} after overwrite operation")
+        log.debug(obj_dump_single)
+        log.info(
+            "Verify change in block size, total size and number of blocks after data removal"
+        )
+        try:
+            log.info(
+                f"{obj_dump_single_stats['blksize']} == {bluestore_min_alloc_size}"
+            )
+            assert obj_dump_single_stats["blksize"] == bluestore_min_alloc_size
+            log.info(
+                f"{obj_dump_single_stats['size']} "
+                f"== {obj_dump_stats['size']} - {(iterations - 1) * 4 * 1048576}"
+            )
+            assert (
+                obj_dump_single_stats["size"]
+                == obj_dump_stats["size"] - (iterations - 1) * 4 * 1048576
+            )
+            log.info(
+                f"{obj_dump_single_stats['blocks']} "
+                f"== {obj_dump_stats['blocks']} "
+                f"- {((iterations - 1) * 4 * 1048576) / bluestore_min_alloc_size}"
+            )
+            assert obj_dump_single_stats["blocks"] == obj_dump_stats["blocks"] - int(
+                ((iterations - 1) * 4 * 1048576) / bluestore_min_alloc_size
+            )
+            l_offset = obj_dump_single_extents[-1]["logical_offset"]
+            length = obj_dump_single_extents[-1]["length"]
+            log.info(f"{l_offset} + {length} == {4 * 1048576}")
+            assert l_offset + length == (4 * 1048576)
+            blocks_l_extent = int(
+                length / obj_dump_single_stats["blksize"]
+            )  # number of blocks in a logical extent
+            log.info(
+                f"{len(obj_dump_single_extents)} == {int(obj_dump_single_stats['blocks']/blocks_l_extent)}"
+            )
+            assert len(obj_dump_single_extents) == int(
+                obj_dump_single_stats["blocks"] / blocks_l_extent
+            )
+            log.info("PASS")
+        except (AssertionError, KeyError) as e:
+            log.info("^FAIL")
+            log.error(f"Failed with exception: {e.__doc__}")
+            log.exception(e)
+            return 1
+    except Exception as e:
+        log.error(f"Execution failed with exception: {e.__doc__}")
+        log.exception(e)
+        return 1
+    finally:
+        check_osd_status(
+            osd_node=acting_osd_node,
+            rados_obj=rados_obj,
+            fsid=fsid,
+            osd_id=acting_osd_id,
+            timeout=timeout,
+            status="up",
+        )
+
+        # Delete the created osd pool
+        if config.get("delete_pool"):
+            rados_obj.detete_pool(pool=pool_name)
+
+    return 0
+
+
+def check_osd_status(
```

Comment:
We can merge this method with : ` change_osd_state ` . Please handle restart command

Reply:
Done, merged the code with `change_osd_state` in `ceph/rados/core_workflows.py`. Verification check is only being performed for start and stop, for everything else, a blind sleep of 5 secs is being used.